src/Generators/euclideangenerators.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    struct Euclidean <: RandomGenerator
+
+A struct representing a graph generator for a Euclidean graph.
+
+### Required Fields
+- `points<:AbstractMatrix{<:Real}}`: a matrix of points
+
+### Optional Parameters
+- `L::Real`: used to bound the `d` dimensional box from which points are selected (default `1.0`)
+- `p::Real`: an edge between vertices `i` and `j` is inserted if `norm(points[:,i]-points[:,j], p) < cutoff` (default `2.0`). If `p=2.0` the standard Euclidean distance will be used.
+- `cutoff::Real`: the distance cutoff (if negative, every edge is harvested)
+- `periodic::Bool`: if `true`, impose periodic boundary conditions in the box ``[0,L]^d`` (default `false`)
+- `return_weights::Bool`: if `true`, constructor should return the weights matrix (default `false`)
+- `return_points::Bool`: if `true`, constructor should return the points matrix (default `false)
```

Comment:
How does that work with `return_weights` and `return_points` will the constructor return a tuple in these cases instead of only a graph?

Reply:
Precisely. Either a 2-tuple or a 3-tuple depending on the options selected.